src/main/lua/exasolvs/query_renderer.lua

```diff
@@ -465,6 +467,31 @@ function M.new (query)
         end
     end
 
+    has_single_group_aggregation = function(select)
+        return select.aggregationType == "single_group"
+    end
+
+    has_aggregate_function = function(select_list)
+        if not select_list then
+            return false
+        end
+        for i = 1, #select_list do
+            if select_list[i].type == "function_aggregate" then
+                return true
+            end
+        end
+        return false
+    end
+
+    local function append_group_by(subquery)
+        if has_single_group_aggregation(subquery) and not has_aggregate_function(subquery.selectList) then
+            append(" GROUP BY 'a'")
```

Comment:
Using a hard-coded alias appears dangerous to me. Especially if there is a chance for a collision with an existing object of that name.

Update: I saw the test. While this fulfills the unit test, it will definitely not be enough in a real-world scenario.